lib/std/assertions.nim

```diff
@@ -30,7 +28,7 @@ when not defined(nimHasSinkInference):
 
 proc raiseAssert*(msg: string) {.noinline, noreturn, nosinks.} =
   ## Raises an `AssertionDefect` with `msg`.
-  sysFatal(AssertionDefect, msg)
+  raise newException(AssertionDefect, msg)
```

Comment:
why is this change here? `sysFatal` afaik has magic meaning for the `--panics` dialect flag

Reply:
`std/assertions` has been moved out of system, but `sysFatal` is a private module inside the system module. So I simply follow the action taken by `std/syncio` https://github.com/nim-lang/Nim/blob/59083e2e48bbf49daaa8d29e233c6f57f33a05ef/lib/std/syncio.nim#L154

Reply:
`raise newException` probably uses `sysFatal` internally.

Reply:
the ability to raise panics that depend on the `panics` flag should be available for everyone I think - ie when I raise _any_ `Defect`, it should honor the panics flag - so the solution here would be that any time anything deriving from `Defect` is raised, it should use the equivalent of `sysFatal` - even if it's a "user-defined Defect"

changing `raiseAssert` significantly changes the way asserts behave, otherwise.

Reply:
> probably uses `sysFatal` internally

so maybe `sysFatal` needs to be removed entirely?

also, why is this a _runtime_ check? the type of the exception is usually known at `raise` time - there should not be runtime overhead for checking this.

Reply:
> changing raiseAssert significantly changes the way asserts behave, otherwise.

It still work like before

```nim
proc foo =
  raiseAssert("124")

proc main =
  try:
    foo()
  except AssertionDefect:
    discard

main()
```

with `panics`, `AssertionDefect` cannot be caught as before.

> also, why is this a runtime check? the type of the exception is usually known at raise time - there should not be runtime overhead for checking this.

That's a separate and pre-existing issue for both `sysFatal` and `raise`.